okhttp/src/main/java/com/squareup/okhttp/HttpUrl.java

```diff
@@ -0,0 +1,294 @@
+  }
+
+  /**
+   * Returns the host address suitable for use with {@link InetAddress#getAllByName(String)}. May
+   * be a regular host name ({@code android.com}), an IPv4 address ({@code 127.0.0.1}), an IPv6
+   * address ({@code ::1}; note that there are no square braces), or an encoded IDN ({@code
+   * xn--n3h.net}).
+   */
+  public String encodedHost() {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Returns the explicitly-specified port if one was provided, or the default port for this URL's
+   * scheme. For example, this returns 8443 for {@code https://square.com:8443/} and 443 for {@code
+   * https://square.com/}.
+   */
+  public int port() {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Returns 80 if {@code scheme.equals("http")}, 443 if {@code scheme.equals("https")} and -1
+   * otherwise.
+   */
+  public static int defaultPort(String scheme) {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Returns the entire path of this URL, encoded for use in HTTP resource resolution. The
+   * returned path is always nonempty and is prefixed with {@code /}.
+   */
+  public String encodedPath() {
+    throw new UnsupportedOperationException();
+  }
+
+  public List<String> pathSegments() {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Returns the query of this URL, encoded for use in HTTP resource resolution. The returned string
+   * may be null (for URLs with no query), empty (for URLs with an empty query) or non-empty (all
+   * other URLs).
+   */
+  public String encodedQuery() {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Returns the first query parameter named {@code name} decoded using UTF-8, or null if there is
+   * no such query parameter.
+   */
+  public String queryParameter(String name) {
+    throw new UnsupportedOperationException();
+  }
+
+  public Set<String> queryParameterNames() {
+    throw new UnsupportedOperationException();
+  }
+
+  public List<String> queryParameterValues(String name) {
+    throw new UnsupportedOperationException();
+  }
+
+  public String queryParameterName(int index) {
+    throw new UnsupportedOperationException();
+  }
+
+  public String queryParameterValue(int index) {
+    throw new UnsupportedOperationException();
+  }
+
+  public String fragment() {
+    throw new UnsupportedOperationException();
+  }
+
+  /**
+   * Returns the URL that would be retrieved by following {@code link} from this URL.
+   *
+   * TODO: explain better.
+   */
+  public HttpUrl resolve(String link) {
+    throw new UnsupportedOperationException();
+  }
+
+  public Builder newBuilder() {
+    return new Builder(this);
+  }
+
+  /**
+   * Returns a new {@code OkUrl} representing {@code url} if it is a well-formed HTTP or HTTPS URL,
+   * or null if it isn't.
+   */
+  public static HttpUrl parse(String url) {
+    throw new UnsupportedOperationException();
+  }
+
+  public static HttpUrl get(URL url) {
```

Comment:
Might not be wise to use `get` here as that is what's used in `Request.Builder` for a GET request. Perhaps `from`?